src/debug/StaticWebAppDebugProvider.ts

```diff
@@ -63,6 +64,9 @@ export class StaticWebAppDebugProvider implements DebugConfigurationProvider {
                 }
 
                 if ((await tryGetApiLocations(context, folder))?.length) {
+                    // make sure Functions extension is installed
+                    await getFunctionsApi(context);
```

Comment:
It might be better to have a custom error for this.  I personally think it could be confusing to debug and see this error `'You must have the "Azure Functions" extension installed to perform this operation.'`  

Isn't the func core tools cli the requirement here?  Does the user actually need the Functions Extension?

Reply:
Yeah we should use a custom error message. And we need Functions because we are relying on the tasks that the Functions extension resolves.

```json
{
  "type": "func",
  "command": "host start",
  "problemMatcher": "$func-node-watch",
  "isBackground": true,
  "dependsOn": "npm build (functions)",
  "options": {
    "cwd": "${workspaceFolder}/api"
  }
}
```